### PR TITLE
Support validate all command

### PR DESCRIPTION
**Pull request by sridhargaddam**
This PR enables validation of cni, connections, deployment, k8s-version and
kube-proxy-mode using a single "validate all" command.

Signed-Off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

**Review by mangelajo — CHANGES_REQUESTED** [review]
As we were talking on slack, the "all" .go file is missing %)

**Comment by mangelajo** [comment]
> As we were talking on slack, the "all" .go file is missing %)

You were quicker than me @sridhargaddam ! :D

File: pkg/subctl/cmd/validate_all.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+		validateCNIInCluster(item.clusterName, submariner)
+		fmt.Println()
+		validateConnectionsInCluster(item.config, item.clusterName)
+		fmt.Println()
+		checkPods(item, submariner, OperatorNamespace)
+		fmt.Println()
+		checkOverlappingCIDRs(item, submariner)
+		fmt.Println()
+		validateKubeProxyModeInCluster(item.config, item.clusterName)
```
Comment (mangelajo) [inline pkg/subctl/cmd/validate_all.go:63]: As we spoke on slack, it'd be nice to add all the firewall check which don't require 2 kubeconfigs.

May be we can add a warning saying that the 2-kubeconfig test needs to be ran separately.

**Review by mangelajo — APPROVED** [review]
Looks good, I'd also add the other validations (firewall with 1-file) can be a separate PR or added to this.

File: pkg/subctl/cmd/validate_all.go (hunk 1 above)
Comment (Jaanki) [inline pkg/subctl/cmd/validate_all.go:63]: add a Println() after this adds a new line before running validation on the next cluster. This beautifies the output.

` ✓ Validating Submariner support for the kube-proxy mode used in cluster "cluster3" 
 ✓ Cluster is not deployed with kube-proxy ipvs mode.

 ✓ Validating Submariner support for the Kubernetes version used in cluster "cluster1"
 ✓ The Kubernetes version meets Submariner's requirements
`

File: pkg/subctl/cmd/validate_all.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+	for _, item := range configs {
+		validateK8sVersionInCluster(item.config, item.clusterName)
+		fmt.Println()
+
+		status.Start(fmt.Sprintf("Retrieving Submariner resource from %q", item.clusterName))
+		submariner := getSubmarinerResource(item.config)
+		if submariner == nil {
+			status.QueueWarningMessage(submMissingMessage)
+			status.End(cli.Success)
```
Comment (Jaanki) [inline pkg/subctl/cmd/validate_all.go:49]: and here as well

` ✓ Validating Submariner support for the Kubernetes version used in cluster "cluster1"
 ✓ The Kubernetes version meets Submariner's requirements

 ✓ Retrieving Submariner resource from "cluster1"
 ⚠ Submariner is not installed

 ✓ Validating Submariner support for the Kubernetes version used in cluster "cluster2"
`

**Review by Jaanki — APPROVED** [review]
Looks good. Just a small UX change. incase you respin. Not urgent though

**Comment by sridhargaddam** [comment]
Thanks for reviewing @mangelajo @Jaanki 
I will merge this PR and work on a new PR to include even firewall validations along with any other changes.

**Pull request merged**